lib/symlink/index.js

```diff
@@ -2,16 +2,35 @@
 
 var through2 = require('through2');
 var fs = require('graceful-fs');
+var path = require('path');
+var valueOrFunction = require('value-or-function');
+
 var prepareWrite = require('../prepare-write');
+var defaultValue = require('../default-value');
+
+var boolean = valueOrFunction.boolean;
 
 function symlink(outFolder, opt) {
+  if (!opt) {
+    opt = {};
+  }
+
+  var rel = defaultValue(false, boolean(opt.relative));
   function linkFile(file, enc, cb) {
     var srcPath = file.path;
     var symType = (file.isDirectory() ? 'dir' : 'file');
     prepareWrite(outFolder, file, opt, function(err) {
       if (err) {
         return cb(err);
       }
+
+      if (rel) {
+        var dstDir = (fs.existsSync(file.path) && symType === 'dir')
+          ? file.path
+          : file.dirname;
+        srcPath = path.relative(dstDir, srcPath);
```

Comment:
I think vinyl has a `file.relative` property that we could use instead of this branch; If it does, you'd be able to just use a ternary when initializing `srcPath` on line 20